### DISCUSSION TITLE
Limiting the local filesystem to a single directory and readonly in memory.

### DISCUSSION THREAD

**Discussion by paulheg** [marcboeker/go-duckdb#331] [Q&A]
Hi,

im currently using duckdb as a in memory database to query `.parquet` files from a local directory.
To increase the security I wanted to configure the following things:

1. **Limiting the file access to a single directory.** 
   - I found the filesystem api: for example `SET disabled_filesystems = 'LocalFileSystem';`, which according to #200 is not exposed.
   - With the `DirFileSystem` it maybe is possible to specify the directory.
2. **Disable write access** in general 
   -  using `access_mode=READ_ONLY` but then I get the error: `Catalog Error: Cannot launch in-memory database in read-only mode`
   -  I created a `dummy.db` to open in readonly, but this did not disable file writing.
   -  I found the option `enable_external_access` the problem with this is, I want to enable read only an not disable both reading and writing.

Can anyone guide me on how to achieve this?
Maybe I can create an empty database file and then fix the read only issue?

Best regards
Paul

**Comment by mlafeldt** [comment]
👋 Thanks for starting this discussion!

This repository has moved to https://github.com/duckdb/duckdb-go

If this topic is still relevant, please continue the conversation at the new location. I'm closing this now as part of the migration.

Please see https://github.com/marcboeker/go-duckdb/issues/565 for more details.
